src/vos/vos_tree.c

```diff
@@ -629,18 +628,17 @@ svt_rec_free_internal(struct btr_instance *tins, struct btr_record *rec,
 		return umem_free(&tins->ti_umm, rec->rec_off);
 
 	/** Find an empty slot for the deferred free */
-	for (i = dth->dth_rsrvd_cnt - 1; i >= 0; i--) {
-		dru = &dth->dth_rsrvds[i];
-		if (dru->dru_scm == NULL)
-			continue;
+	for (i = 0; i < dth->dth_deferred_cnt; i++) {
+		rsrvd_scm = dth->dth_deferred[i];
```

Comment:
We do not need to find out the free slot one by one. Because we have already created N "vos_rsrvd_scm" slots when init the DTX handle, so just directly use "rsrvd_scm = dth->dth_deferred[dth_op_seq];", nobody will share using such slot with current sub-modification.

Reply:
Good point.  I guess you can never have more undo's than the number of submodifications.